packages/Takeover/src/components/FocusTrap.js

```diff
@@ -0,0 +1,36 @@
+import PropTypes from "prop-types";
+import FocusTrapLibrary from "focus-trap-react";
+
+// properties copy from https://github.com/davidtheclark/focus-trap
+
+const propTypes = {
+  /** A function that will be called when the focus trap activates. */
+  onActivate: PropTypes.func,
+  /** A function that will be called when the focus trap deactivates */
+  onDeactivate: PropTypes.func,
+
+  /** By default, when a focus trap is activated the first element in the focus trap's tab order will receive focus. With this option you can specify a different element to receive that initial focus. Can be a DOM node, or a selector string (which will be passed to document.querySelector() to find the DOM node), or a function that returns a DOM node. */
+  initialFocus: PropTypes.oneOfType([PropTypes.node, PropTypes.string, PropTypes.func]),
+
+  /**
+    By default, an error will be thrown if the focus trap contains no elements in its tab order. With this option you can specify a fallback element to programmatically receive focus if no other tabbable elements are found. For example, you may want a popover's <div> to receive focus if the popover's content includes no tabbable elements. Make sure the fallback element has a negative tabindex so it can be programmatically focused. The option value can be a DOM node, a selector string (which will be passed to document.querySelector() to find the DOM node), or a function that returns a DOM node.
+    escapeDeactivates {boolean}: Default: true. If false, the Escape key will not trigger deactivation of the focus trap. This can be useful if you want to force the user to make a decision instead of allowing an easy way out.
```

Comment:
Can we just remove this line?

Reply:
Probably, you missed the purpose of `FocusTrap` component.

As you can see it always renders `null`. The only one purpose of this component is to declare propTypes. When someone uses `<Takeover>` component, they have ability to override focus trap options by passing this component as `<Takeover>` child.

All props passed to `<Takeover.FocusTrap>` will be [extracted](https://github.com/acl-services/paprika/blob/fbd96f0b0397e3af5e69d3bdae6c8d01508b8dfe/packages/Takeover/src/Takeover.js#L63-L75) and [applied to actual focus trap component](https://github.com/acl-services/paprika/blob/fbd96f0b0397e3af5e69d3bdae6c8d01508b8dfe/packages/Takeover/src/Takeover.js#L99).

And I guess we should keep an ability to override/add any `focus-trap-react` option.

Reply:
True, but having the `escapeDeactivates` prop here, in the comments, seems strange.  I think a link to the online documentation for the `focus-trap-react` library, maybe directly to the options, would be more helpful?